### PR TITLE
read: warn on invalid options

### PR DESCRIPTION
**Pull request by davidvgalbraith**
Check the ES mapping each read and use that knowledge to trigger a warning if a program uses an unknown `timeField` or filters on or reduces by an analyzed or unknown field. 

@demmer @VladVega

### CODE REVIEW

File: lib/read.js [hunk 1]
```diff
@@ -65,6 +67,50 @@ class ReadElastic extends AdapterRead {
...
+                    _.each(index_mapping, (props_object, type) => {
+                        var properties = props_object.properties;
+                        this._warn_if_missing(properties, this.timeField, index, type);
+
+                        this.filtered_fields.forEach((field) => {
+                            this._warn_if_analyzed(properties, field, index, type);
+                            this._warn_if_missing(properties, field, index, type);
+                        });
+
+                        var reduce_by_fields = this.es_opts.aggregations &&
```
Comment (VladVega) [inline lib/read.js]: nit, just a preference here so only change if you agree: 
I like `var y = a  || b` but  `var y = a && c  || b` isn't as clear to read as it is clever. 
Maybe use this instead:
```
var reduce_by_fields = [];
if (this.es_opts.aggregations) {
     reduce_by_fields = this.es_opts.aggregations.grouping;
}
```

Comment (VladVega) [inline lib/read.js]: Actually nm, maybe just add parens? `(a && b) || c`

File: lib/elastic.js [hunk 2]
```diff
@@ -59,17 +60,20 @@ function init(config) {
...
-    } else {
-        var endpoint = _.findWhere(clients, {id: id});
-        if (!endpoint) {
-            throw new Error('invalid id: ' + id);
-        }
+function _instance_for_id(id) {
+    if (id === undefined) { return instances[0]; }
 
-        return endpoint.client;
+    var endpoint = _.findWhere(instances, {id: id});
```
Comment (VladVega) [inline lib/elastic.js]: Consider using either `instance` or `endpoint` but not both?

**Comment by davidvgalbraith** [comment]
Updated.

**Comment by VladVega** [comment]
My one other concern: Should they be warnings or errors that stop the program immediately?

**Comment by davidvgalbraith** [comment]
Warnings! You still get some data, it just might not be what you expect unless you're super-ES-savvy. This is what Kibana does.

**Comment by VladVega** [comment]
Ok +1.

**Pull request merged**